plugin/src/main/java/org/opensearch/ml/rest/mcpserver/RestMcpConnectionMessageStreamingAction.java

```diff
@@ -122,31 +147,136 @@ protected void prepareRequestInternal(final String path, final String sessionId,
-                                channel.sendResponse(new BytesRestResponse(RestStatus.OK, "OK"));
+                    .doOnSuccess(x -> {
+                        String requestBody = x.utf8ToString();
+                        ActionListener<GetResponse> listener = ActionListener.wrap(r -> {
+                            if (r.isExists()) {
+                                String nodeId = String.valueOf(r.getSourceAsMap().get("node_id"));
+                                DiscoveryNode node = clusterService.state().getNodes().getNodes().get(nodeId);
+                                if (node == null) {
+                                    log.error("The node:{} is no longer in the current cluster, can not handle the mcp request", r.getId());
+                                    channel
+                                        .sendResponse(
+                                            new BytesRestResponse(
+                                                channel,
+                                                new IllegalStateException(
+                                                    "Session no longer exists as corresponding node crashed, please recreate a new session in client side"
+                                                )
+                                            )
+                                        );
+                                } else {
+                                    ActionListener<AcknowledgedResponse> actionListener = ActionListener.wrap(y -> {
+                                        if (y.isAcknowledged()) {
+                                            log.debug("MCP request has been dispatched to corresponding node and handled successfully!");
+                                            if (requestBody.contains("notifications/initialized")) {
+                                                log
+                                                    .debug(
+                                                        "Starting to send OK response for notifications/initialized request in coordinator node"
+                                                    );
+                                                channel.sendChunk(createInitializedNotificationRes());
+                                            }
+                                        }
+                                    },
+                                        e -> {
+                                            log
+                                                .error(
+                                                    "MCP request has been dispatched to corresponding node but peer node failed to handle it",
+                                                    e
+                                                );
+                                        }
+                                    );
+                                    if (clusterService.localNode().getId().equals(nodeId)) {
+                                        client
```

Comment:
Do we need this ? I see same code in else block

Reply:
If you check the first commit, you can see such same code 
```
                                        client
                                            .execute(
                                                MLMcpMessageAction.INSTANCE,
                                                new MLMcpMessageRequest(sessionId, requestBody),
                                                actionListener
                                            );
```

Reply:
But I see you removed such code in the second commit for `(clusterService.localNode().getId().equals(nodeId))`. I think such code not needed for local node case, right ?

Reply:
At first I thought you mean the `clusterService.localNode().getId().equals(nodeId)` in else. I removed the part you mentioned, it's not needed.